cirq-core/cirq/transformers/analytical_decompositions/two_qubit_to_fsim.py

```diff
@@ -67,9 +67,10 @@ def decompose_two_qubit_interaction_into_four_fsim_gates(
         mapped_gate = ops.FSimGate(-fsim_gate.exponent * np.pi / 2, 0)
     else:
         mapped_gate = fsim_gate
-    if not 3 / 8 * np.pi <= abs(mapped_gate.theta) <= 5 / 8 * np.pi:
+    theta, phi = cast(float, mapped_gate.theta), cast(float, mapped_gate.phi)
```

Comment:
can this be done by an assertion instead?  I think we were trying to use assertions instead of casts in cases where speed wasn't as important.  (Maybe it is important here, not sure but don't think it is critical). 

For the ones in other files which deal with unitary probably good to use cast.

Reply:
done